File: CPP/5_Majority_Element.cpp
/*
Given an array nums of size n, return the majority element.

The majority element is the element that appears more than ⌊n / 2⌋ times. You may assume that the majority element always exists in the array.
*/

int majorityElement(vector<int>& nums) {
    int candidate = 0;
    int count = 0;
    for(int ele:nums){
        if(count == 0){
            candidate = ele;
            count = 1;
        } else if (ele == candidate) {
            count++;
        } else {
            count--;
        }
    }
    return candidate;
}

File: Python/5_Majority_Element.py
'''
Given an array nums of size n, return the majority element.

The majority element is the element that appears more than ⌊n / 2⌋ times. You may assume that the majority element always exists in the array.
'''

def majorityElement(nums):
    candidate = 0;
    count = 0;
    for num in nums:
        if count == 0:
            candidate = num
            count = 1
        elif candidate == num:
            count += 1
        else:
            count -= 1
    return candidate